src/feature/dircache/dircache.c

```diff
@@ -1438,6 +1441,27 @@ handle_get_networkstatus_bridges(dir_connection_t *conn,
   return 0;
 }
 
+/** Helper function for GET the bandwidth file used for the next vote */
+static int
+handle_get_next_bandwidth(dir_connection_t *conn,
+                          const get_handler_args_t *args)
+{
+  (void)args;
+  log_debug(LD_DIR, "Getting next bandwidth.");
+  const or_options_t *options = get_options();
+  if (options->V3BandwidthsFile) {
+    int lifetime = 60;
+    char *bandwidth =  read_file_to_str(options->V3BandwidthsFile, 0, NULL);
+    size_t len = strlen(bandwidth);
```

Comment:
This explodes if `bandwidth` is NULL.

Reply:
glups.
Question, is there any advantage on passing an struct stat to read_file_to_str and get the length from st_size instead of getting the length with strlen?

Reply:
Not really:
https://github.com/torproject/tor/blob/537092cdbb7f4be0e6d68f4e5d65ca2a403375f9/src/lib/fs/files.c#L601-L617

read_file_to_str() makes sure that the sizes are consistent. If they're not, it fails:
https://github.com/torproject/tor/blob/537092cdbb7f4be0e6d68f4e5d65ca2a403375f9/src/lib/fs/files.c#L700-L710